proposed/http-message.md

```diff
@@ -332,9 +505,11 @@ interface IncomingRequestInterface extends MessageInterface
      * purposes of how duplicate query parameters are handled, and how nested
      * sets are handled.
      *
-     * @return array
+     * @param array $query Array of query string arguments, typically from
+     *     $_GET.
+     * @return void
      */
-    public function getQueryParams();
+    public function setQueryParams(array $query);
```

Comment:
The body parameter is a bit different though, because in cases where `$_POST` is populated, the body tends to be empty. I actually felt that the cookie-related stuff should never have made it in there, best to just let those people parse it out :)

One thing I didn't realize though, is that QueryParams is only a part of the ServerRequest, and not the generic request. That makes this not as bad, and more logical imho so I'll happily drop this.